### PR TITLE
Refactor table test in controller_test

### PR DESCRIPTION
**Pull request by mlguerrero12**
Run cases of the table test as subtests. This causes IDEs such as Goland to recognize each case separately. It is then possible to run each case directly. It also improves logging.

### CODE REVIEW

File: controller/controller_test.go [hunk 1]
```diff
@@ -897,20 +895,7 @@ func TestControllerMutation(t *testing.T) {
...
-		}
-
-		if t.Failed() {
-			// Don't run more test cases if we've already failed, to
-			// keep the output readable.
-			break
-		}
-
-		// Shuffle the input vector, and run again.
-		for x := range tests {
```
Comment (fedepaol) [inline controller/controller_test.go:909]: why this part was removed? I don't think it's part of this change and I guess we still want it.
If a commit says "move to subtests", it should do that and nothing more.

File: controller/controller_test.go [hunk 2]
```diff
@@ -897,20 +895,7 @@ func TestControllerMutation(t *testing.T) {
 					panic("bad wanted IP in loadbalancer status")
 				}
 			}
-		}
-
-		if t.Failed() {
```
Comment (fedepaol) [inline controller/controller_test.go:902]: why this was removed?

File: controller/controller_test.go (hunk 1 above)
Comment (mlguerrero12) [reply controller/controller_test.go:909]: Right, out of the scope of this PR. Will add it back.

**Comment by fedepaol** [comment]
lgtm

**Pull request merged**